osu.Game/Extensions/DrawableExtensions.cs

```diff
@@ -43,5 +45,23 @@ public static ScheduledDelegate BeginKeyRepeat(this IKeyBindingHandler handler,
         /// <returns>The delta vector in Parent's coordinates.</returns>
         public static Vector2 ScreenSpaceDeltaToParentSpace(this Drawable drawable, Vector2 delta) =>
             drawable.Parent.ToLocalSpace(drawable.Parent.ToScreenSpace(Vector2.Zero) + delta);
+
+        public static SkinnableInfo CreateSerialisedInformation(this Drawable component) => new SkinnableInfo(component);
+
+        public static void ApplySerialisedInformation(this Drawable component, SkinnableInfo info)
+        {
+            // todo: can probably make this better via deserialisation directly using a common interface.
+            component.Position = info.Position;
+            component.Rotation = info.Rotation;
+            component.Scale = info.Scale;
+            component.Anchor = info.Anchor;
+            component.Origin = info.Origin;
+
+            if (component is Container container)
+            {
+                foreach (var child in info.Children)
+                    container.Add(child.CreateInstance());
+            }
+        }
```

Comment:
I couldn't decide whether these should be `DrawableExtensions` or `SkinnableComponentExtensions`. Open to moving them. The hope is that eventually `IDrawable` can be expanded to cover all the requirements of this PR so we don't need to ever cast to `Drawable`, though.